### PR TITLE
test: Support random-generated-struct testing in mapping functions

### PR DESCRIPTION
**Pull request by c-warren**
**What changed?**

Adds a generic function to support testing mapping functions with randomly generated data. 

**Why?**

Related to https://github.com/cadence-workflow/cadence/issues/7611

We map from one type to another in many places in our code base - external (thrift/grpc) types to internal types, internal types to persistence types, etc. When mapping from one type to another there are multiple error scenarios that could occur in a round trip:

- neither mapper implements a field
- the mapper implementations are incompatible
- both mappers implement a field correctly
- both mappers implement conversion incorrectly but in a reversible way (e.g they both do a rot13 of an [a-z] string)

Using default 'testdata' does test for this but can miss edge cases (e.g the first case is completely missed). We already have fuzz testing within cadence however it isn't widely used - partly because the existing pattern requires significant boilerplate code. This will reduce the amount of boilerplate required for a simple mapping function. 

See the implementation within the cadence-go-client: https://github.com/cadence-workflow/cadence-go-client/pull/1457

**How did you test it?**

The implementation of the generic test function was tested via implementing it within the Schedule mappers, which are currently using the existing pattern. 

```bash
go test ./common/types/mapper/...
```

**Potential risks**

None

**Release notes**

None

**Documentation Changes**

None

### CODE REVIEW

File: common/types/mapper/testutils/fuzz_mapper.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+						// For struct elements, pass the address to ensure modifications work
+						// For pointer elements, pass the interface directly
+						if elem.Kind() == reflect.Struct && elem.CanAddr() {
+							clearFieldsIf(elem.Addr().Interface(), shouldClear)
+						} else {
+							clearFieldsIf(elem.Interface(), shouldClear)
+						}
+					}
+				}
+			case reflect.Map:
```
[the remark this answers is not in the document]
Comment (c-warren) [reply common/types/mapper/testutils/fuzz_mapper.go:290]: I'm ok with this for consistency with the cadence-go-client implementation, but it's worth creating an issue to remove it/test it further.

File: common/types/mapper/testutils/fuzz_mapper.go [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+	}
+}
+
+// WithTimeFuzzers adds custom fuzzers for time.Time and time.Duration
+// to ensure safe timestamp bounds and proper UTC normalization
+func WithTimeFuzzers() FuzzOption {
+	return WithCustomFuncs(
+		// time.Time with safe bounds (Unix epoch to year 2100) and UTC normalization
+		func(t *time.Time, c fuzz.Continue) {
+			if c.Intn(10) < 3 {
```
Comment (Shaddoll) [inline common/types/mapper/testutils/fuzz_mapper.go:99]: Are these arbitrary numbers you chose?

**Review by Shaddoll — APPROVED** [review]

File: common/types/mapper/testutils/fuzz_mapper.go (hunk 2 above)
Comment (c-warren) [reply common/types/mapper/testutils/fuzz_mapper.go:99]: I copied them from the existing implementation to keep the same behavior. 
```
func(t *time.Time, c fuzz.Continue) {
			if c.Intn(10) < 3 {
				*t = time.Time{}
				return
			}
			*t = time.Unix(c.Int63n(4102444800), c.Int63n(1e9)).UTC()
		},
		func(d *time.Duration, c fuzz.Continue) {
			if c.Intn(10) < 3 {
				*d = 0
				return
			}
			*d = time.Duration(c.Int63n(int64(24 * time.Hour)))
		},
```

In the go client I just used the safe timestamps:
```
func(d *gogo.Duration, c fuzz.Continue) {
			d.Seconds = c.Int63n(MaxDurationSeconds)
			d.Nanos = 0 // Thrift mapping truncates nanoseconds, so set to 0 for consistency
		},
```
